### PR TITLE
Add CRS for for install and upgrade tanzu-addons-manager package

### LINKED ISSUE #1585: Update addons manager package with new CRDs and RBAC
(This is used to request new product features)

**Describe the feature request**

- Add all Config CRDs to addons manager carvel package
- Add TanzuclusterbootstrapTemplate and Tanzuclusterbootstrap CRD
- Add appropriate RBAC's updating kubebuilder annotations if needed
- Update BOM to reference the new package

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [x] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [x] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by ggpaue**
Signed-off-by: Guanpeng Gao <gguanpeng@vmware.com>

### What this PR does / why we need it
Add CRS for for install and upgrade tanzu-addons-manager package
### Which issue(s) this PR fixes

Fixes #1585 

### Release note

```release-note

```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/providers/ytt/02_addons/management-packages/addons-manager/tanzu-addons-manager_overlay.lib.yaml [hunk 1]
```diff
@@ -9,6 +9,6 @@ metadata:
   #@overlay/match missing_ok=True
   annotations:
     ext.packaging.carvel.dev/ytt-data-values-overlays: "tanzu-addons-manager-data-values"
-    tkg.tanzu.vmware.com/addon-type: "addons-management/tanzu-addons-manager"
+    tkg.tanzu.vmware.com/addon-type: "packages/management-package"
```
Comment (blc1996) [inline pkg/v1/providers/ytt/02_addons/management-packages/addons-manager/tanzu-addons-manager_overlay.lib.yaml:12]: Question: Do all the management packages share the same `addon-type`?

File: pkg/v1/tkg/client/upgrade_addon.go [hunk 2]
```diff
@@ -182,11 +182,6 @@ func (c *TkgClient) DoUpgradeAddon(regionalClusterClient clusterclient.Client, /
 			clusterClient = currentClusterClient
 		case "addons-management/kapp-controller":
 			crsDisabledAddon = true
-		case "addons-management/tanzu-addons-manager":
```
Comment (blc1996) [inline pkg/v1/tkg/client/upgrade_addon.go:185]: Are we assuming case "packages/management-package" will update the addons manager package after the removal of case "addons-management/tanzu-addons-manager"?
Question: Will "packages/management-package" have multiple packages?

File: pkg/v1/providers/ytt/02_addons/management-packages/addons-manager/tanzu-addons-manager_overlay.lib.yaml (hunk 1 above)
Comment (ggpaue) [reply pkg/v1/providers/ytt/02_addons/management-packages/addons-manager/tanzu-addons-manager_overlay.lib.yaml:12]: @blc1996 yes, they use the same addon-type: "packages/management-package"

**Comment by alfredthenarwhal** [comment]
> @blc1996:  /test
Commit: 9bdbc8f

Tests failed! Build no: 1474

File: pkg/v1/providers/ytt/02_addons/management-packages/addons-manager/tanzu-addons-manager_package.lib.yaml [hunk 3]
```diff
@@ -42,10 +42,7 @@ metadata:
 spec:
   serviceAccountName: tanzu-addons-package-sa
   packageRef:
-    #! PackageName is available in TKR BOM under the particular path listed below. We do not have to validate the key existence
-    #! is because YTT will throw error if any struct is None or keys are missing, we expect to see those error and templating
-    #! will be exit immediately.
-    refName: #@ get_bom_data_for_tkr_name().addons["tanzu-addons-manager"].packageName
+    refName: addons-manager.tanzu.vmware.com
```
Comment (yharish991) [inline pkg/v1/providers/ytt/02_addons/management-packages/addons-manager/tanzu-addons-manager_package.lib.yaml:45]: i think you need to add `tkg.tanzu.vmware.com/addon-type: "packages/management-package"` annotation to all the resources in `tanzuaddonsmanagerpackageyaml` function

File: pkg/v1/tkg/client/upgrade_addon.go (hunk 2 above)
Comment (ggpaue) [reply pkg/v1/tkg/client/upgrade_addon.go:185]: yeah all the management packages will use this addon type

File: pkg/v1/providers/ytt/02_addons/management-packages/addons-manager/tanzu-addons-manager_package.lib.yaml (hunk 3 above)
Comment (ggpaue) [reply pkg/v1/providers/ytt/02_addons/management-packages/addons-manager/tanzu-addons-manager_package.lib.yaml:50]: annotations added

**Pull request closed without merging**